### ISSUE TITLE
Policy on upgrading mapbox-gl-js

### ISSUE THREAD

**Issue by kturney** [kturney/ember-mapbox-gl#18]
@danielthall @chriswhong @allthesignals what are y'alls thoughts on how ember-mapbox-gl's version should change when upgrading the underlying mapbox-gl-js library?

It seems excessive to bump our major version every time they release a major version, but if we don't we are potentially transitively making breaking changes.

Alternatively, we could have ember-mapbox-gl depend on `"mapbox-gl": "*"` and then consuming apps would be in charge of adding a direct dependency on a specific version of mapbox-gl, which ember-mapbox-gl would then pick up. But then we run the risk of the resolved version having some incompatibilities with our bindings.

Or maybe we wing it until they hit 1.0?

**Comment by allthesignals** [comment]
@miguelcobain, could you weigh in on this? Didn't you have a similar issue with Leaflet pre-1.0 and post-1.0?  

I think `"mapbox-gl": "*"` is fine until 1.0, that's when breaking API changes will be introduced, if they are. I'll wait for other thoughts.

**Comment by kturney** [comment]
So far most of their `0.X` bumps have had breaking changes.

**Comment by miguelcobain** [comment]
When leaflet 1 was released I major bumped ember-leaflet. But that wasn't a problem. Leaflet doesn't major bump that often and I took the chance to make some breaking changes.

In ember-leaflet I have ember-try scenarios that still test the library with leaflet 0.7. This is very important for some users because some leaflet plugins are still just compatible with this leaflet version.

The strategy we take allows consumers to specify the leaflet version ember-leaflet uses.
We specify the minimum leaflet version as a peerDependency: https://github.com/miguelcobain/ember-leaflet/blob/master/package.json#L64-L66

And we add the npm package afterInstall in https://github.com/miguelcobain/ember-leaflet/blob/master/blueprints/ember-leaflet/index.js#L18

This way users can specify what version they want.
